src/prototype_creep.js

```diff
@@ -84,21 +84,53 @@ Creep.prototype.handle = function() {
     if (this.memory.last === undefined) {
       this.memory.last = {};
     }
-    let last = this.memory.last;
-    this.memory.last = {
-      pos1: this.pos,
-      pos2: last.pos1,
-      pos3: last.pos2,
-    };
+    if(this.fatigue === 0){
+      let last = this.memory.last;
+      this.memory.last = {
+        pos1: this.pos,
+        pos2: last.pos1,
+        pos3: last.pos2,
+        pos4: last.pos3,
+        pos5: last.pos4,
+      };
+    }
+    this.handleStuck();
   }
 };
-
 Creep.prototype.isStuck = function() {
-  return this.memory.last !== undefined &&
-    this.memory.last.pos3 !== undefined &&
-    this.pos.isEqualTo(this.memory.last.pos3.x, this.memory.last.pos3.y);
+  if(this.memory.last !== undefined && this.memory.last.pos5 !== undefined && this.pos.isEqualTo(this.memory.last.pos5.x, this.memory.last.pos5.y)){
+    if(!this.memory.stuckCount){
+      this.memory.stuckCount = 0;
+    }
+    if(!this.memory.stuckx){
+      this.memory.stuckx = this.memory.last.pos1.x;
+    }
+    if(!this.memory.stucky){
+      this.memory.stucky = this.memory.last.pos1.y;
+    }
+    if(this.memory.stuckx == this.memory.last.pos1.x && this.memory.stucky == this.memory.last.pos1.y){
+      return true;
+    } else {
+      return false;
+    }
+  } else {
+    return false;
+  };
+};
+Creep.prototype.handleStuck = function() {
+  if(this.isStuck() && (this.role == 'harvester' || this.role == 'carry' || this.role == 'planer' || this.role == 'repairer' || this.role == 'scout')){
+    this.memory.stuckCount = this.memory.stuckCount + 1;
+    this.log('Stuck! - ' + this.memory.stuckCount);
+    if(this.memory.stuckCount > 100){
+      this.log('UnStuck!')
```

Comment:
Make the message more prominent and maybe with some debug output. We shouldn't come to this situation.